test/Microsoft.Azure.ServiceBus.UnitTests/MessageInterOpTests.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.ServiceBus.UnitTests
+{
+    using Microsoft.Azure.ServiceBus.Core;
+    using Microsoft.Azure.ServiceBus.Extensions;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Runtime.Serialization;
+    using System.Threading.Tasks;
+    using Xunit;
+
+    public class MessageInterOpTests
+    {
+        public static IEnumerable<object> TestSerializerPermutations => new object[]
+        {
+            new object[] { new DataContractBinarySerializer(typeof(Book)) },
+            new object[] { new DataContractSerializer(typeof(Book)) },
+        };
+
+        public static IEnumerable<object> TestEnd2EndEntityPermutations => new object[]
+        {
+            new object[] { TestConstants.NonPartitionedQueueName },
+            new object[] { TestConstants.PartitionedQueueName}
+        };
+
+        [Theory]
+        [MemberData(nameof(TestSerializerPermutations))]
+        [DisplayTestMethodName]
+        void RunSerializerTests(XmlObjectSerializer serializer)
+        {
+            Book book = new Book("contoso", 1, 5);
```

Comment:
I prefer writing the full name but this is test class, so not to worried. Will change